frontend/src/metabase/dashboard/actions/tabs.ts

```diff
@@ -128,7 +128,7 @@ export const tabsReducer = createReducer<DashboardState>(
           const newTab = getDefaultTab({
             tabId,
             dashId,
-            name: t`Page ${prevTabs.length + 1}`,
+            name: t`Page ${prevTabs.filter(t => !t.isRemoved).length + 1}`,
```

Comment:
There was a bug where new tabs would have the wrong default name if you previously deleted some tabs in the same edit mode session